catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/activity/rtl/MultilingualTest.java

```diff
@@ -0,0 +1,138 @@
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.uiespresso.activity.rtl;
+
+import android.support.test.runner.AndroidJUnit4;
+
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.ui.Multilingual;
+import org.catrobat.catroid.uiespresso.util.BaseActivityInstrumentationRule;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.Locale;
+
+import static android.support.test.espresso.Espresso.onData;
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.assertThat;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+
+import static junit.framework.Assert.assertTrue;
+
+import static org.catrobat.catroid.uiespresso.util.UiTestUtils.getResourcesString;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.hasToString;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.startsWith;
+
+@RunWith(AndroidJUnit4.class)
+public class MultilingualTest {
+	private static final Locale ARABICLOCALE = new Locale("ar");
+	private static final Locale URDULOCALE = new Locale("ur");
+	private static final Locale FARSILOCALE = new Locale("fa");
+	private static final Locale DEUTSCHLOCALE = Locale.GERMAN;
+	@Rule
+	public BaseActivityInstrumentationRule<Multilingual> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(Multilingual.class);
+
+	@Before
+	public void setUp() throws Exception {
+		baseActivityTestRule.launchActivity(null);
+	}
+
+	@Test
+	public void testChangeLanguageToDeutsch() {
+		onData(hasToString(startsWith(DEUTSCHLOCALE.getDisplayName(DEUTSCHLOCALE)))).perform(click());
+		assertThat("current Locale is not Deutsch", Locale.getDefault().getDisplayLanguage(), is(DEUTSCHLOCALE.getDisplayLanguage()));
+		String buttonProgramsName = getResourcesString(R.string.main_menu_programs);
+		String buttonContinueName = getResourcesString(R.string.main_menu_continue);
+		String buttonNewName = getResourcesString(R.string.main_menu_new);
+		onView(withId(R.id.main_menu_button_continue))
```

Comment:
you have right, I can use that with all buttons except ContinueButton because the return value will be the string R.string.main_menu_continue + the name of the program, to avoid that error I use
`onView(withId(R.id.main_menu_button_continue))
				.check(matches(withText(containsString(buttonContinueName))));`